### PR TITLE
Added the ability to clear sizing information from Contour so that the next render re-sizes the visualization(s)

### PR DESCRIPTION
**Pull request by wasbridge**
Patch allows us to:

var contour = new Contour({ el:'.myChart' })
                   .pie([1,2,3])
                   .render()
             
var onResize = function(e) {
    contour.resize().render();
}

window.addEventListener('resize', onResize);

This is useful so that developers can force Contour to re-measure everything when the element which Contour is rendering into changes size

### CODE REVIEW

File: src/scripts/core/contour.js [hunk 1]
```diff
@@ -343,6 +343,45 @@
...
+        *     var onResize = function(e) {
+        *          contour.resize().render();
+        *     }
+        *
+        *     window.addEventListener('resize', onResize);
+        *
+        * @function resize
+        *
+        */
+        resize: function(width, height) {
```
Comment (jaimedp) [inline src/scripts/core/contour.js]: I think this approach works for most of the cases, can you add a @param entry for the parameters in the comments?

File: src/scripts/core/axis/linear-scale-axis.js [hunk 2]
```diff
@@ -31,11 +31,12 @@
                 this._scale = d3.scale.linear().domain(this._domain);
                 if(this.options.xAxis.min == null && this.options.xAxis.max == null)
                     this._scale.nice();
-                this._setRange();
             } else {
                 this._scale.domain(this._domain);
             }
 
+            this._setRange();
```
Comment (jaimedp) [inline src/scripts/core/axis/linear-scale-axis.js]: Do we need to do this for the other axis types? (time/ordinal/etc)

Comment (wasbridge) [inline src/scripts/core/axis/linear-scale-axis.js]: The y axes respond well -- (can't test log axis b/c I can't figure out how to get Contour to use it)
The y axes did need some updating

**Comment by wasbridge** [comment]
I totally screwed up this branch when introducing the optimization changes -- I am going to resubmit with the changes seperate

**Pull request closed without merging**